acp/internal/controller/task/task_controller.go

```diff
@@ -844,130 +799,109 @@ func generateRandomString(n int) (string, error) {
 	return string(ret), nil
 }
 
-// createHumanContactRequest builds the request payload for sending to a response URL
-func createHumanContactRequest(agentName string, result string) ([]byte, error) {
-	// Use agent name as runId
-	runID := agentName
-	// Generate a secure random string for callId
-	callID, err := generateRandomString(7)
-	if err != nil {
-		return nil, fmt.Errorf("failed to generate secure random string: %w", err)
+// Defined for testing purposes
+var newHumanLayerClientFactory = humanlayer.NewHumanLayerClientFactory
+
+// sendFinalResultViaHumanLayerAPI sends the final task result using the HumanLayer API client
+func (r *TaskReconciler) sendFinalResultViaHumanLayerAPI(ctx context.Context, task *acp.Task, result string) error {
+	logger := log.FromContext(ctx)
+
+	if task.Spec.BaseURL == "" || task.Spec.ChannelTokenFrom == nil {
+		logger.Info("Skipping result notification, BaseURL or ChannelTokenFrom not set")
+		return nil
 	}
-	spec := humanlayerapi.NewHumanContactSpecInput(result)
-	input := humanlayerapi.NewHumanContactInput(runID, callID, *spec)
-	return json.Marshal(input)
-}
 
-// isRetryableStatusCode determines if an HTTP status code should trigger a retry
-func isRetryableStatusCode(statusCode int) bool {
-	return statusCode >= 500 || statusCode == 429
-}
+	// Get the channel token from the secret
+	var secret corev1.Secret
+	if err := r.Get(ctx, client.ObjectKey{
+		Namespace: task.Namespace,
+		Name:      task.Spec.ChannelTokenFrom.Name,
+	}, &secret); err != nil {
+		return fmt.Errorf("failed to get channel token secret: %w", err)
+	}
 
-// sendFinalResultToResponseURL sends the final task result to the specified URL
-// It includes retry logic for transient errors and better error categorization
-func (r *TaskReconciler) sendFinalResultToResponseURL(ctx context.Context, task *acp.Task, result string) error {
-	logger := log.FromContext(ctx)
-	logger.Info("Sending final result to responseURL", "responseURL", task.Spec.ResponseURL)
+	channelToken := string(secret.Data[task.Spec.ChannelTokenFrom.Key])
+	if channelToken == "" {
+		return fmt.Errorf("channel token is empty in secret %s/%s key %s",
+			task.Namespace, task.Spec.ChannelTokenFrom.Name, task.Spec.ChannelTokenFrom.Key)
+	}
 
-	// Create the request body
-	jsonData, err := createHumanContactRequest(task.Spec.AgentRef.Name, result)
+	// Create HumanLayer client factory with the BaseURL
+	clientFactory, err := newHumanLayerClientFactory(task.Spec.BaseURL)
```

Comment:
its really hard to figure out by reading this PR where BaseURL gets used and if we're appending the `/humanlayer/v1` inside the client. (we shouldn't be)